src/main/java/gov/cabinetoffice/gap/adminbackend/listeners/SchemeUpdateListener.java

```diff
@@ -0,0 +1,24 @@
+package gov.cabinetoffice.gap.adminbackend.listeners;
+
+import gov.cabinetoffice.gap.adminbackend.entities.SchemeEntity;
+import gov.cabinetoffice.gap.adminbackend.models.AdminSession;
+import gov.cabinetoffice.gap.adminbackend.utils.HelperUtils;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import java.time.Instant;
+
+@Slf4j
+public class SchemeUpdateListener {
+
+    @PrePersist
+    @PreUpdate
+    private void beforeAnyUpdate(SchemeEntity scheme) {
```

Comment:
Not entirely sure how UpdateListeners operate but would a PostUpdate not be preferable in case of any exceptions thrown during updating causing a mismatch between updated time and scheme state?